### PR TITLE
New: Support legacy and editable auto captions

### PR DESCRIPTION
**Pull request by jeremypress**
tests on the way

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jeremypress has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #465

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentSidebar/Skills/SidebarSkills.js [hunk 1]
```diff
@@ -28,86 +28,125 @@ type Props = {
...
-        case SKILLS_KEYWORD:
-            return SKILLS_TARGETS.KEYWORDS.CARD;
-        case SKILLS_FACE:
-        case SKILLS_TIMELINE:
-            return SKILLS_TARGETS.FACES.CARD;
-        case SKILLS_TRANSCRIPT:
-            return SKILLS_TARGETS.TRANSCRIPTS.CARD;
-        default:
-            return '';
+class SidebarSkills extends React.PureComponent<Props> {
```
Comment (priyajeet) [inline src/components/ContentSidebar/Skills/SidebarSkills.js]: Do all this in SkillsSidebar.js

**Comment by priyajeet** [comment]
Ok, here is your another issue. Skills sidebar tab currently opens by default, so your code gets executed. If the skills sidebar is no longer the default in the future, then your code won't reach.

**Comment by jeremypress** [comment]
@priyajeet True, we could either move this farther out to contentSidebar. I've settled on this approach because doing anything in preview would be blocked by the read token/ The token used by T3

**Comment by priyajeet** [comment]
Another issue, the viewer load event may get fired before the sidebar is rendered.

**Comment by jeremypress** [comment]
Before I add the event listener I can check to see if the viewer is loaded, and call the method right away. I can move the initial setup to contentsidebar, and then keep the update on edit in the component.

**Pull request closed without merging**